guides/Postgres Query Optimization + Diagnostic Tooling.md

```diff
@@ -0,0 +1,68 @@
+ The performance difference is not significant, but according to the sources reviewed in the literature study, modifying the use of * is expected to enhance performance.
+ 
+ ### Filtering Rows with Subqueries and Indexes:
+ Filtering data before performing complex operations like GROUP BY or JOIN can enhance query performance by minimizing the amount of data that needs to be processed. Utilizing subqueries or other techniques to filter data is often effective in reducing query execution time and improving database efficiency. <br><br>
+&nbsp;&nbsp;   ![](/.gitbook/assets/images/Picture6.png)   
+ <br>
+ 
+ &nbsp;&nbsp; ![](/.gitbook/assets/images/Picture7.png)
+ ### WHERE Clause:
+ To optimize the WHERE Clause, we can use OR instead of IN when there is no index on the filtering column. The IN statement matches a column value to a list of values, and technically, it should execute faster than OR. However, in some cases, using OR instead of IN may not improve performance significantly, but it is a way to test if the query runs faster. <br><br>
+ &nbsp;&nbsp;   ![](/.gitbook/assets/images/Picture8.png)   
+ <br>
+ 
+ &nbsp;&nbsp; ![](/.gitbook/assets/images/Picture9.png)
+ 
+### Optimize Joins:
+To enhance query performance, optimizing joins is essential as they are expensive operations. Substituting joins with subqueries whenever possible is one way to optimize them. Additionally, using the JOIN ON syntax instead of the WHERE clause can help the optimizer produce better execution plans. Filtering records of large tables before joining them with other tables can also lead to substantial performance improvements.
+
+These are some additional factors that should be taken into consideration when designing a Postgres query for improved performance.  
+*	Temporary tables can slow down execution but can also avoid the need for ORDER BY operations.  
+*	Some operations may prevent the query from using indexes, so it's important to understand these peculiarities.  
+*	The order of tables in the FROM statement can affect JOIN ordering, particularly when joining more than five tables.  
+*	Index-only scans are faster than full table scans, but index access can be slower when query selectivity is high.  
+*	Views can result in inefficient queries.  
+*	Use DISTINCT only when necessary.  
+*	Minimize the use of subqueries, especially correlated subqueries.  
+*	Long queries are not helped by indexes, but instead can be optimized by minimizing the number of full table scans and reducing the result size as soon as possible.  
+*	Materialized views are useful for speeding up execution time if having fully up-to-date data is not a priority for the query.  
+
+## Conclusion:
+In conclusion, optimizing Postgres queries is crucial for improving performance and reducing costs. Using diagnostic tools like EXPLAIN and EXPLAIN ANALYZE can help identify potential issues. Techniques such as using indexes, filtering columns, and optimizing joins can significantly improve 
```

Comment:
I think the end of this paragraph might have gotten lost. Mind reviewing and adding the finishing thoughts?

Reply:
I will polish the ending part.

Reply:
added missing part.